src/pages/cocktail/[id]/index.js

```diff
@@ -0,0 +1,142 @@
+
+function Redirect({ to }) {
+  const router = useRouter();
+
+  useEffect(() => {
+    router.push(to);
+  }, [router, to]);
+
+  return null;
+}
+
+export default function CocktailPage({ drinkData }) {
+  const [shouldRedirect, setShouldRedirect] = useState(false);
+  const { isFallback } = useRouter();
+
+  if (shouldRedirect) {
+    return <Redirect to="/" />;
+  }
+
+  return (
+    <Container maxWidth="md">
+      {isFallback ? (
+        <>
+          <Typography variant="h6" color="primary">
+            Loading...
+          </Typography>
+          <br />
+          <CircularProgress />
+        </>
+      ) : (
+        <>
+          <Header subtitle={`Cocktail ${drinkData.strDrink}`} />
+          <Card
+            style={{
+              display: 'flex',
+              alignItems: 'center',
+              backgroundColor: '#FFEFD5',
+              marginTop: '30px',
+            }}
+          >
+            <CardMedia
+              component="img"
+              sx={{ height: 140 }}
+              title={`cocktail ${drinkData.strDrink}`}
+              image={`${drinkData.strDrinkThumb}/preview`}
+            />
+            <CardContent>
+              <Typography
+                style={{ display: 'inline-block' }}
+                variant="h6"
+                color="primary"
+              >
+                Best served in:
+              </Typography>{' '}
+              <Typography
+                style={{ display: 'inline-block' }}
+                variant="body1"
+                color="initial"
+              >
+                {drinkData.strGlass}{' '}
+              </Typography>
+              <br />
+              <br />
+              <Typography variant="h6" color="primary">
+                {' '}
+                Instructions:{' '}
+              </Typography>
+              <Typography variant="body1" color="initial">
+                <span data-cy="instructions">{drinkData.strInstructions}</span>{' '}
+              </Typography>
+            </CardContent>
+          </Card>
+          <div style={{ display: 'flex', justifyContent: 'center' }}>
+            <Box m={4}>
+              <span data-cy="back-btn">
+                <Button
+                  onClick={() => setShouldRedirect(true)}
+                  style={{ padding: '20px', color: '#B8860B' }}
+                >
+                  Go back to Main
+                </Button>
+              </span>
+            </Box>
+          </div>
+          <br />
+        </>
+      )}
+    </Container>
+  );
+}
+
+// Static
+export async function getStaticPaths() {
+  const res = await fetch(
+    `https://www.thecocktaildb.com/api/json/v1/1/filter.php?c=Cocktail`,
+  );
+  const data = await res.json();
+  const paths = data.drinks.map((drink) => ({
+    params: { id: drink.idDrink.toString() },
+  }));
+  return { paths, fallback: true };
+}
+
+export async function getStaticProps({ params }) {
+  const { id } = params;
+  const details = await fetch(
```

Comment:
let's abstract this http calls to a different folder/file, to make it independent of the implementation and easy to test.